cranelift/codegen/src/isa/aarch64/abi.rs

```diff
@@ -708,6 +721,29 @@ impl ABIMachineSpec for AArch64MachineDeps {
         insts
     }
 
+    fn gen_memcpy(
+        call_conv: isa::CallConv,
+        dst: Reg,
+        src: Reg,
+        size: usize,
+    ) -> SmallVec<[Self::I; 8]> {
+        let mut insts = SmallVec::new();
+        insts.push(Inst::gen_move(writable_xreg(0), dst, I64));
+        insts.push(Inst::gen_move(writable_xreg(1), src, I64));
+        insts.extend(Inst::load_constant(writable_xreg(2), size as u64).into_iter());
+        insts.push(Inst::Call {
```

Comment:
It is important that the stack space for the outgoing arguments is not clobbered other than the stack space for the argument this should write. Using `emit_vm_call` makes it much harder to prevent accidentally doing this I think.

Reply:
Yep, we need to be careful in exactly how we set up args and emit this call here, so I think it's best to open-code it.